### PR TITLE
Smarter layout picking in CostCalculator

### PR DESCRIPTION
**Pull request by pnowojski**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
 import com.facebook.presto.spi.Constraint;
 import com.facebook.presto.spi.statistics.Estimate;
 import com.facebook.presto.spi.statistics.TableStatistics;
+import com.facebook.presto.spi.type.Type;
```
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java:27]: Put in commit message why the change.

File: presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java [hunk 2]
```diff
@@ -82,10 +88,12 @@ public CoefficientBasedCostCalculator(Metadata metadata)
             extends PlanVisitor<Map<PlanNode, PlanNodeCost>, Void>
     {
         private final Session session;
+        private final Map<Symbol, Type> types;
 
-        public Visitor(Session session)
+        public Visitor(Session session, Map<Symbol, Type> types)
         {
             this.session = session;
+            this.types = types;
```
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java:96]: ImmutableMap.copyOf?

File: presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java [hunk 3]
```diff
@@ -106,10 +114,18 @@ private void visitChildren(PlanNode node, Map<PlanNode, PlanNodeCost> context)
         @Override
         public Void visitFilter(FilterNode node, Map<PlanNode, PlanNodeCost> context)
         {
-            visitChildren(node, context);
+            boolean fullyEnforced = false;
+            if (node.getSource() instanceof TableScanNode && !((TableScanNode) node.getSource()).getLayout().isPresent()) {
```
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java:118]: It seems like you may remove right conjunct.

**Review by losipiuk — APPROVED** [review]

File: presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java (hunk 2 above)
Comment (pnowojski) [reply presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java:96]: fixed

File: presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java (hunk 3 above)
Comment (pnowojski) [reply presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java:118]: fixed
